### ISSUE TITLE
v2 cli binary: only darwin includes the embedded cli-ext, but not linux and windows

### ISSUE THREAD

**Issue by panmenghan** [hasura/graphql-engine#7509]
### Version Information

Server Version: v2.0.8
CLI Version (for CLI related issue): v2.0.8

### Environment

windows 10

### What is the expected behaviour?
```
Applying metadata...
```

### What is the current behaviour?

```
WARN Unable to find an embedded cli-ext. So trying to fetch it from CDN
WARN Tip: --cli-ext-path can be used for setting up cli-ext from local file system
Applying metadata...
```

### How to reproduce the issue?

`hasura metadata apply`

### Screenshots or Screencast

v2.0.8 cli binaries: 
![v2 0 8](https://user-images.githubusercontent.com/51240264/132183436-e606ec0a-0c7a-4952-9330-a5a9c875bb64.png)

v2.0.0-beta2 cli binaries:
![v2 0 0-beta 2](https://user-images.githubusercontent.com/51240264/132183405-d221549f-8b3b-4cc2-bc74-d1b711afe6c2.png)

### Any possible solutions?

From the screenshots, I guess only darwin(after v2.0.0-beta2) includes the embedded cli-ext, but not linux and windows.
Can make all the cli binaries to include the embedded cli-ext?

**Comment by scriptnull** [comment]
Hi @panmenghan, thanks for reporting :pray: 

The reason for the difference in binary sizes is because of the way we perform compression on the Hasura CLI binaries. The current setup uses [upx](https://upx.github.io) to compress the binaries for linux and windows platforms. However, compression is disabled for the macOS (darwin) binaries as it seemed to cause problems related to code signing. Hence, we stopped compressing the darwin binaries after `v2.0.0-beta.2`.

We confirmed that cli-ext is correctly embedded for linux and darwin binaries. This issue helped us uncover a bug that seems to not embed cli-ext binary on the windows platform alone. Luckily, we had a fallback logic to fetch cli-ext from CDN, which lead to the warning message in the issue description.

We just landed a fix for this via https://github.com/hasura/graphql-engine/commit/2329c16ce909051f1226fe4be8f72267bdf30e8b and windows cli should use an embedded cli-ext instead of fetching it every time from the CDN once the fix is released in the upcoming release.
